py3status/modules/stopwatch.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+"""
+A simple stopwatch.
+
+This is a very basic stopwatch. You can start, pause and reset the
+stopwatch.
+
+Button 1 starts/pauses the stopwatch.
+Button 2 resets stopwatch.
+
+Configuration parameters:
+    format: display format for this module (default 'Stopwatch {stopwatch}')
+
+Format placeholders:
+    {stopwatch} display hours:minutes:seconds
+
+@author Jonas Heinrich
+
+SAMPLE OUTPUT
+{'full_text': 'Stopwatch 0:01:00'}
+
+running
+[
+    {'full_text': 'Stopwatch '},
+    {'color': '#00FF00', 'full_text': '0'},
+    {'full_text': ':'},
+    {'color': '#00FF00', 'full_text': '00'},
+    {'full_text': ':'},
+    {'color': '#00FF00', 'full_text': '54'},
+]
+
+paused
+[
+    {'full_text': 'Stopwatch '},
+    {'color': '#FFFF00', 'full_text': '0'},
+    {'full_text': ':'},
+    {'color': '#FFFF00', 'full_text': '00'},
+    {'full_text': ':'},
+    {'color': '#FFFF00', 'full_text': '54'},
+]
+"""
+
+from time import time
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    format = 'Stopwatch {stopwatch}'
+
+    def post_config_hook(self):
+        self.running = False
+        self.time_start = None
+        self.time_state = None
+        self.color = None
+        self.paused = False
+
+    def stopwatch(self):
+
+        def make_2_didget(value):
+            value = str(value)
+            if len(value) == 1:
+                value = '0' + value
+            return value
+
+        if self.running:
+            t = int(time() - self.time_start)
+        else:
+            if self.time_state:
+                t = self.time_state
+            else:
```

Comment:
I think we might be able to use `elif` here to make it a bit more cleaner.

Reply:
Or leave this alone and put `cached_until` stuffs in there.

Reply:
done